### PR TITLE
Don't preprend . to service name if package not set

### LINKED ISSUE #450: support .proto files without package declarations
I don't even know if this is possible, but I follow the quick-start-scala and used the same proto file to generate a Node.js client and it didn't work. Getting the error:  Received http2 header with status: 404

I was able to do this using pure ScalaPB and Node.js.

Any thoughts?

### PR DESCRIPTION
**Pull request by chbatey**
Fixes #450

### CODE REVIEW

**Comment by chbatey** [comment]
Is there an easy way to add this to the interop tests? I didn't see the protos in there

File: codegen/src/main/scala/akka/grpc/gen/scaladsl/Service.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ object Service {
     Service(
       fileDesc.scalaPackageName,
       serviceClassName,
-      fileDesc.getPackage + "." + serviceDescriptor.getName,
+      (if (fileDesc.getPackage.isEmpty) "" else fileDesc.getPackage + ".") + serviceDescriptor.getName,
```
Comment (helena) [inline codegen/src/main/scala/akka/grpc/gen/scaladsl/Service.scala]: Maybe we should consider adding a new `FileDescriptorOps` for the logic in one versus two, duplicated.

Comment (helena) [reply codegen/src/main/scala/akka/grpc/gen/scaladsl/Service.scala]: To provide one convention.

**Comment by raboof** [comment]
> Is there an easy way to add this to the interop tests? I didn't see the protos in there

The interop tests mainly test against https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md and get the `.proto` from the upstream jar.

We could:
* widen the `interop tests` to also include other tests that require code generation
* introduce a new subproject for other tests that require code generation
* only test features that require code generation in 'scripted' tests in the sbt-plugin subproject

'1' sounds fine to me. Getting the code generated might be as simple as adding a proto in that subproject under `src/main/proto`.

**Comment by chbatey** [comment]
> '1' sounds fine to me. Getting the code generated might be as simple as adding a proto in that subproject under src/main/proto.

Sounds good will do that

**Pull request merged**